### DISCUSSION TITLE
Wazuh Agent log collector - what's next?

### DISCUSSION THREAD

**Discussion by thanhsang2606** [wazuh/wazuh#14875] [Q&A] [answered]
Hi my friends,

I'm using the newest Wazuh 4.3.7, installed Wazuh Agent in an Linux server, every function, every module work perfectly, great work for Wazuh team.
![image](https://user-images.githubusercontent.com/104063390/190104221-e0d8b755-9853-49a5-876c-6b31701d1350.png)

Then I followed [this guide](https://documentation.wazuh.com/current/user-manual/capabilities/log-data-collection/how-it-works.html#remote-syslog) to configured for Wazuh Agent collect some of the custom logs in **/var/logs/postgresql & /var/logs/gvm** without using the **rsyslog**, the logs in Wazuh Agent is in the attach, seems work correctly, but nothing in the **archive.json and archive.log** in Wazuh Manager (already **enabled log all** in ossec.conf), I did enable the debug mode for wazuh-logcollector in both Wazuh Manager and Wazuh Agent, but not thing usefull.

Can you guys tell me which one I missed or what's thing next?
[Wazuh_Manager_Log.txt](https://github.com/wazuh/wazuh/files/9564334/Wazuh_Manager_Log.txt)
[Wazuh_Agent_Config.txt](https://github.com/wazuh/wazuh/files/9564335/Wazuh_Agent_Config.txt)
[Wazuh_Agent_Log.txt](https://github.com/wazuh/wazuh/files/9564336/Wazuh_Agent_Log.txt)
[Wazuh_Manager_Config.txt](https://github.com/wazuh/wazuh/files/9564337/Wazuh_Manager_Config.txt)

**Comment by mdiego92** [comment]
Hello thanhsang2606,

It seems that the log collection is working properly. We will need to confirm if there are any decoders and rules working for your logs be received as events in Wazuh Dashboard.

Could you please share with us some log entries to analyze them and help you with it?

Thank you very much.

**Comment by thanhsang2606** [comment]
[redis-server-openvas.log](https://github.com/wazuh/wazuh/files/9570610/redis-server-openvas.log)
[openvas.log](https://github.com/wazuh/wazuh/files/9570611/openvas.log)
[postgresql-14-main.log](https://github.com/wazuh/wazuh/files/9570612/postgresql-14-main.log)
Yes, thanks for your reply, seems the decoders and the rules are working fine, I'm used the Wazuh Ruleset test and as you can see, it can decord and trigger the log from Postgre. Beside at, I have attach all the original logs I want to check out, please tell me if you need any additional information.
<img width="1250" alt="image" src="https://user-images.githubusercontent.com/104063390/190281518-666a0145-da58-4ec5-903c-414aff88d081.png">

**Reply by mdiego92** [reply]
Hello [thanhsang2606](https://github.com/thanhsang2606),

It seems that your logs are not matching the decoders. For example, if your PostgreSQL logs don't have brackets **[ ]**, we will need to change the configuration for that. Could you please follow these steps?

Change the `<localfile>` configuration in your `ossec.conf` file to use the `postgresql_log` decoder:

```
<localfile>
    <location>/var/log/postgresql/postgresql-14-main.log</location>
    <log_format>postgresql_log</log_format>
</localfile>
```

To override a default decoder you shouldn't update the original file, since any changes in the `/var/ossec/ruleset/decoders` folder will be lost in the update process. The override procedure is:

Exclude the original decoder file from the OSSEC loading list. To do this, use the tag` <decoder_exclude>` in the `ruleset` section from the `ossec.conf` file. Thus, the specified decoder will not be loaded from the default decoder folder, and the decoder file saved in the user folder will be loaded instead.

`<decoder_exclude>ruleset/decoders/0225-postgresql_decoders.xml</decoder_exclude>
`

Add the following decoder to the `/var/ossec/etc/decoders/local_decoder.xml` file:

```
<decoder name="postgresql_log">
  <prematch>^\d\d\d\d-\d\d-\d\d \S+ \w+ </prematch>
  <regex offset="after_prematch">^\S+ \S+@\S+ (\w+): </regex>
  <order>status</order>
</decoder>
```

Bear in mind that, if updates to the public Wazuh Ruleset include changes to the decoder you overrode, they will not apply to you since you are no longer loading that decoder file from the standard location that gets updated.

Here's the documentation section about overriding a decoder:
https://documentation.wazuh.com/current/user-manual/ruleset/custom.html#changing-an-existing-decoder

**Finally, you will have to restart the Wazuh Manager to apply these changes.**

A similar procedure should be done with the other log files you have.

Articles for reference:
https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/localfile.html
https://documentation.wazuh.com/current/user-manual/ruleset/ruleset-xml-syntax/rules.html
https://documentation.wazuh.com/current/user-manual/ruleset/ruleset-xml-syntax/decoders.html

Please let me know if you have any remaining questions!

Thank you very much.

**Comment by thanhsang2606** [comment]
**Hi @mdiego92**, I followed your instruction such below steps:
- Excluded postgresql decoder in /var/ossec/etc/ossec.conf
- Add the custom postgresql decoder to /var/ossec/etc/decoders/local_decoder.xml
- Changed <localfile> format log from system to postgresql_log
- Enabled logall to see any postgresql logs from the archive.log
- Restarted Wazuh Manager
But it seems still doesn't work, I saw the log is analyzing the postgre ossec.log file, but nothing shown in Security Module and the archive.log.
Attached below new log and config files (have change the file suffix to .txt for upload):
[ossec_conf.txt](https://github.com/wazuh/wazuh/files/9596732/ossec_conf.txt)
[ossec_log.txt](https://github.com/wazuh/wazuh/files/9596733/ossec_log.txt)
[local_decoder_xml.txt](https://github.com/wazuh/wazuh/files/9596734/local_decoder_xml.txt)
[archives.log](https://github.com/wazuh/wazuh/files/9597047/archives.log)
The Ruleset Test worked after changed:
<img width="966" alt="image" src="https://user-images.githubusercontent.com/104063390/190981677-457cda9b-5922-47d8-aa9a-40757a7a56d1.png">

**Comment by mdiego92** [comment, marked as answer]
Hello @thanhsang2606,

In the example you sent us, the log file is properly identified with the rules and decoders. Since the alert level is 0 you won't see any events in your security events section. By default, the minimum alert level for the alerts to appear is 3.

For this log entry, you should be receiving alerts:
`2022-09-12 10:17:52.282 UTC [39047] postgres@postgres FATAL:  password authentication failed for user "postgres"
`
```
**Phase 1: Completed pre-decoding.
       full event: '2022-09-12 10:17:52.282 UTC [39047] postgres@postgres FATAL:  password authentication failed for user "postgres"'
       timestamp: '(null)'
       hostname: 'wazuh-server'
       program_name: '(null)'
       log: '2022-09-12 10:17:52.282 UTC [39047] postgres@postgres FATAL:  password authentication failed for user "postgres"'

**Phase 2: Completed decoding.
       decoder: 'postgresql_log'
       status: 'FATAL'

**Phase 3: Completed filtering (rules).
       Rule id: '50512'
       Level: '9'
       Description: 'PostgreSQL: Database authentication failure.'

```

You could review the following ruleset file https://github.com/wazuh/wazuh-ruleset/blob/master/rules/0300-postgresql_rules.xml to check what events you will be receiving or analyze if you will need to create specific custom rules.

Please let us know if you have any remaining questions.

Thank you for the update on the issue!

**Comment by thanhsang2606** [comment]
Yeah, thanks so much.
It's worked for me, but only work when I copied the `postgresql-14-main.log `from the target server to the Wazuh Manager server and add more `<localfile>` in `ossec.conf`  file of Wazuh Manager to perform analyst right in Wazuh Server, if the log file just stay in the Postgre server (which install the Wazuh Agent), it doesn't works even the `ossec.log` in the Wazuh Agent shown `Analyzing file: '/var/log/postgresql/postgresql-14-main.log'`
